metrics/metrics-tests/src/main/java/org/apache/james/metrics/tests/RecordingMetric.java

```diff
@@ -56,6 +57,8 @@ public void remove(int i) {
 
     @Override
     public long getCount() {
-        return value.get();
+        return Optional.of(value.get())
+            .filter(counter -> counter > 0)
```

Comment:
As this is a testing utility we should fail (throw runtime exception) to catch error managements on metrics early?

Reply:
then better I will do it in .decrement() and remove()

Reply:
Ok for recording metric